### PR TITLE
Added CI Workflow

### PR DESCRIPTION
**Pull request by nagarjun-repala**
# Resolves

- [AR-1830](https://team-1624093970686.atlassian.net/browse/AR-3453)

## Changes

- Added CI

# Checklist

- [X] The changes have been tested locally

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      run: npm install        
+    - name: Run linters
+      run: npm run lint:ts
+    - name: Build
+      run: npm run compile
+    - name: Test
+      run: npm test
+    - name: Coverage report
+      run: npm run coverage     
+    # - name: Upload test coverage
```
Comment (jdk2588) [inline .github/workflows/ci.yml]: Why is this portion commented? If not in use, why can't we remove it?

**Review by SauravKanchan — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (DudeWhoCode) [reply .github/workflows/ci.yml]: Or we can integrate codecov right? That’s part of acceptance critera

**Review by DudeWhoCode — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (nagarjun-repala) [reply .github/workflows/ci.yml]: I don't have permission to add CodeCov token in repo secrets. Need help from someone to add the secret and test

Comment (jdk2588) [reply .github/workflows/ci.yml]: Ok, then please ask the respective dev to add CodeCov token.

Comment (nagarjun-repala) [reply .github/workflows/ci.yml]: Done

**Review by DudeWhoCode — APPROVED** [review]

**Pull request merged**